CMakeLists.txt

```diff
@@ -1,18 +1,5 @@
-cmake_minimum_required (VERSION 3.16.3)
-
-foreach(p
-    CMP0048 # OK to clear PROJECT_VERSION on project()
-    CMP0054 # CMake 3.1
-    CMP0056 # export EXE_LINKER_FLAGS to try_run
-    CMP0057 # Support no if() IN_LIST operator
-    CMP0063 # Honor visibility properties for all targets
-    CMP0067 # Honor language standard in try_compile() source file signature
-    CMP0077 # Allow option() overrides in importing projects
-    )
-  if(POLICY ${p})
-    cmake_policy(SET ${p} NEW)
-  endif()
-endforeach()
+# Require CMake 3.10. If available, use the policies up to CMake 3.22.
+cmake_minimum_required (VERSION 3.10...3.22)
```

Comment:
https://google.github.io/benchmark/dependencies.html is pretty clear about why we support 3.16.3

Reply:
I guess I am asking you to reconsider.  A number of other Google projects are perfectly capable of making forward progress (AFAICT that is the *why* in that document) while supporting CMake >= 3.10.  And it is not like this project was using any CMake > 3.16 features.  In fact, it seems this project uses more CMake < 3.0 idioms than CMake >= 3 idioms.

In any case, this is your project.  If you cannot support CMake 3.10, we will just have to keep using the older versions.

Reply:
@dmah42 I also think this policy may be worth reconsidering. Abseil, Protobuf, and gRPC (among others) have published a [C++ support policy](https://opensource.google/documentation/policies/cplusplus-support) that allows us to move forward while also allowing each project to remain compatible with eachother. More and more Google projects are adopting these policies. I think it would be good if Google benchmark considering following suite.

Reply:
i didn't intend this to be the end of the conversation.  it was a little brusque (sorry) but my intent was to ensure that we understand the implications of opening up to older cmake versions.

i remember we had to go past 3.7 to support some features at some point, but we may have just skipped right over 3.10.

Reply:
Thanks @dmah42 ! I'll let @coryan follow up w/ the PR, but I wanted to also point out that at the top of https://opensource.google/documentation/policies/cplusplus-support there's a link to [this table](https://github.com/google/oss-policies-info/blob/main/foundational-cxx-support-matrix.md), which lists the specific versions of things that we support. It makes it a bit easier to see which versions the policies implies. In this case, we can see that the table is showing CMake >= 3.10, which is where Carlos was getting these versions.

Reply:
yep, i saw. i think we might want to just point at this rather than track our own dependencies, as long as we're true to it (as in my other comment).